nixos/modules/config/sysctl.nix

```diff
@@ -59,5 +59,19 @@ in
     # Disable YAMA by default to allow easy debugging.
     boot.kernel.sysctl."kernel.yama.ptrace_scope" = mkDefault 0;
 
+    # Dynamic since kernel v5.12
+    # https://github.com/torvalds/linux/commit/ac7b79fd190b02e7151bc7d2b9da692f537657f3
+    boot.kernel.sysctl."fs.inotify.max_user_instances" =
+      if lib.versionAtLeast config.boot.kernelPackages.kernel.version "5.12"
```

Comment:
you mean versionOlder than `5.11` ? (one arithmetic operation away from the magic number `5.12`)

Reply:
If there the is a bugfix version (independent if that actually would happen) it can break in theory. So I'm against that semantic.

Reply:
no, means
```suggestion
      if (config.boot.kernelPackages.kernel.kernelOlder "5.12")
```

The `kernelOlder` and `kernelAtLeast` have been around since 20.09, so they should be available

Reply:
Thanks, it should be `kernalAtLeast` :+1: I'll use those, instead.

Reply:
Done in https://github.com/NixOS/nixpkgs/compare/cbe2715c72c66cea81815e31a642d7c852607b81..c366d5ec80c58130feff4f3e98ca9a8b03c33e1f

Reply:
Yea, had the logic reversed when I wrote it :)

Reply:
It's nice, so if there should be ever a kernel specific twist to it, this will be used.